### PR TITLE
✨adds cli to list the content of state directories in dy-sidecar

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Allows for people trying to save the state of the sidecar figure out where the data is present, without know where the underlying defined service has it's state directories.

### 🦸 How would batman/robin use it?

- Check if the service has any data to save
```command
simcore-service-dynamic-sidecar state-list-dirs
```

- Save the data in the service
```command
simcore-service-dynamic-sidecar state-save
```

**NOTE: you also have autocomplete in the terminal**

## Related issue/s

- https://github.com/ITISFoundation/osparc-issues/issues/638

## Checklist

- [x] Unit tests for the changes exist

### CODE REVIEW

File: services/dynamic-sidecar/tests/unit/test_cli.py [hunk 1]
```diff
@@ -39,17 +39,27 @@ def mock_nodeports(mocker: MockerFixture) -> None:
 # TESTS
 
 
+def test_list_state_dirs(
+    cli_runner: CliRunner, mock_rabbitmq: None, mock_data_manager: None
+):
+    result = cli_runner.invoke(main, ["state-list-dirs"])
+    assert result.exit_code == 0, result.stdout
```
Comment (pcrespov) [inline services/dynamic-sidecar/tests/unit/test_cli.py:46]: MINOR: use https://docs.python.org/3/library/os.html#os.EX_OK

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/cli.py [hunk 2]
```diff
@@ -52,6 +52,22 @@ def _print_highlight(message: str) -> None:
...
+    async def _async_state_list_dirs() -> None:
+        app = await _setup_app_for_task_execution()
+
+        mounted_volumes: MountedVolumes = app.state.mounted_volumes
+
+        for state_path in mounted_volumes.state_paths:
+            state_path_content = list(state_path.glob("*"))
+            typer.echo(f"Entries in {state_path}: {state_path_content}")
+
+    asyncio.get_event_loop().run_until_complete(_async_state_list_dirs())
```
Comment (pcrespov) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/cli.py:68]: shoudn't we use ``asyncio.run`` now? SEE https://docs.python.org/3/library/asyncio-task.html#asyncio.run

**Review by pcrespov — APPROVED** [review]

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/cli.py (hunk 2 above)
Comment (GitHK) [reply services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/cli.py:68]: 👍 Thanks, was searching for this the last time but I could not find it!

**Review by sanderegg — APPROVED** [review]
nice 👍

**Pull request merged**